### PR TITLE
deploy agent from kustomize yaml

### PR DESCRIPTION
**Pull request by philbrookes**
- Follow the steps in the readme to deploy the controller.
- Follow the steps in the readme to deploy the agent.
- Create sample application in workload cluster: `kubectl create -f samples/syncer/application.yaml`
- Watch logs of agent for: certificate secret in place for  host adding dns endpoints
- Curl the host.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -83,7 +83,22 @@ When deploying the multi cluster traffic controller using the make targets the f
...
     (export $(cat ./controller-config.env | xargs) && export $(cat ./aws-credentials.env | xargs) && make build install run
     ```
-## 3. Run the ingress agent locally
+
+## 3. Deploy the agent
+1. Update the secret in `config/agent/secret.yaml` with the correct credentials for the control plane. (**N.B.** The server should remain `https://mctc-control-plane-control-plane:6443`)
+
+1. Build the agent image and load it into the workload cluster
+    ```sh
+    export KUBECONFIG=./hack/kubeconfigs/mctc-workload-1.kubeconfig
```
Comment (sergioifg94) [inline README.md:92]: Is there any difference between this and `kubectl config use-context kind-mctc-workload-1`? Just out of curiosity

Comment (philbrookes) [reply README.md:92]: just that you can set different values in different terminals. Using `kubectl config use-context` changes the context in all terminals (I like to have one terminal connected to the control-plane and one connected to the workload cluster)

Comment (sergioifg94) [reply README.md:92]: Oh yes I hadn't thought of that! That's definitely better

File: README.md [hunk 2]
```diff
@@ -83,7 +83,22 @@ When deploying the multi cluster traffic controller using the make targets the f
     ```sh
     (export $(cat ./controller-config.env | xargs) && export $(cat ./aws-credentials.env | xargs) && make build install run
     ```
-## 3. Run the ingress agent locally
+
+## 3. Deploy the agent
+1. Update the secret in `config/agent/secret.yaml` with the correct credentials for the control plane. (**N.B.** The server should remain `https://mctc-control-plane-control-plane:6443`)
```
Comment (sergioifg94) [inline README.md:88]: I wonder if we could automate this step in a similar way to how it's done here:
https://github.com/Kuadrant/multi-cluster-traffic-controller/blob/main/hack/.argocdUtils

Not necessarily as part of this PR though

Comment (philbrookes) [reply README.md:88]: yeah, I'd really love to get this automated, and out of the path of accidental commits.

**Review by sergioifg94 — APPROVED** [review]
Verified all works as expected

**Pull request merged**